src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -320,56 +319,58 @@ class AgoraBusiness(permissionsDataSource: PermissionsDataSource) extends LazyLo
       excludedFields = Seq.empty[String])
 
     val methodsAndConfigsFuture = findWithIds(AgoraEntity(), Some(projection), Seq(AgoraEntityType.Configuration, AgoraEntityType.Workflow), username)
+    // find public-ness
+    val aliasesOwnersPermissionsFuture = permissionsDataSource.inTransaction(_.aePerms.listAliasesOwnersPermissions)
 
     methodsAndConfigsFuture flatMap { methodsAndConfigsSnapshots =>
+      aliasesOwnersPermissionsFuture map { aliasesOwnersPermissions =>
 
-      val groupedMC = methodsAndConfigsSnapshots.groupBy(_.entityType)
-      val methodSnapshots = groupedMC.getOrElse(Some(AgoraEntityType.Workflow), Seq.empty[AgoraEntity])
-      val configSnapshots = groupedMC.getOrElse(Some(AgoraEntityType.Configuration), Seq.empty[AgoraEntity])
+        val methodSnapshots = methodsAndConfigsSnapshots.filter(_.entityType.exists(_ == AgoraEntityType.Workflow))
+        val configSnapshots = methodsAndConfigsSnapshots.filter(_.entityType.exists(_ == AgoraEntityType.Configuration))
 
-      // group/count config snapshots by methodId
-      val configCounts:Map[Option[ObjectId],Int] = configSnapshots
-          .groupBy(_.methodId)
-          .map { kv => kv._1 -> kv._2.size}
+        // group/count config snapshots by methodId
+        val configCounts: Map[Option[ObjectId], Int] = configSnapshots.groupBy(_.methodId) map {
+          case (objectIdOption, agoraEntities) => objectIdOption -> agoraEntities.size
+        }
```

Comment:
Since this is independent of map on `aliasesOwnersPermissionsFuture `, can this be moved before map statement on 326?

Reply:
Refactored and added comments.